src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -473,18 +477,24 @@ def bad_move(self, order: str) -> bool:
 
     def update_allies_and_foes(self) -> None:
         power_stance = self.stance.stance[self.power_name]
+        neighbors = neighboring_opps(self.game, self.power_name, self.opponents)
 
         self.allies = [
             pow for pow in self.opponents if power_stance[pow] > self.ally_threshold
         ]
         self.foes = [
-            pow for pow in self.opponents if power_stance[pow] <= self.enemy_threshold
+            pow for pow in neighbors if power_stance[pow] <= self.enemy_threshold
         ]
+
+        # disable aggressive move filtering against allies if there are no neighboring enemies
+        if not self.foes:
+            self.is_filtering_moves = False
+        else:
+            self.is_filtering_moves = True
```

Comment:
This can be a single statement:

```suggestion
        self.is_filtering_moves = bool(self.foes)
```

As a bigger change, I personally think that `foes` should be left as it was before because we still want other behavior to treat non-adjacent powers as enemies. Instead, we can decide whether to filter moves by looping over `foes`:

```python
self.is_filtering_moves = any(pow in neighbors for pow in self.foes)
```

Reply:
Currently, with this change we only consider enemies from the neighboring powers in order to "limi[t] the candidates to neighboring powers ensures that the new enemy is one we are able to attack". If we do that, then all of this code will only affect move filtering

Reply:
I think we want it to only affect move filtering. We shouldn't ignore other foes, even if we can't attack them.

Reply:
sounds good!